storage/src/mmr/journaled.rs

```diff
@@ -236,6 +236,58 @@ impl<B: Blob, E: RStorage<B> + Clock + Metrics, H: Hasher> Mmr<B, E, H> {
         self.mem_mmr.add(h, element)
     }
 
+    /// Pop the given number of elements from the tip of the MMR assuming they exist, and otherwise
+    /// return Empty or ElementPruned errors.
+    pub async fn pop(&mut self, mut leaves_to_pop: usize) -> Result<(), Error> {
+        // See if the elements are still cached in which case we can just pop them from the in-mem
+        // MMR.
+        loop {
+            match self.mem_mmr.pop() {
+                Ok(_) => {
+                    leaves_to_pop -= 1;
+                    if leaves_to_pop == 0 {
+                        return Ok(());
+                    }
+                }
+                Err(Error::ElementPruned(_)) => break,
+                Err(Error::Empty) => {
+                    return Err(Error::Empty);
+                }
+                _ => unreachable!(),
+            }
+        }
+        assert!(leaves_to_pop > 0);
```

Comment:
I think this should be above the loop (as well)? The current code could panic with underflow if 0 is provided.

Reply:
This has been removed and replaced with the leaves_to_pop == 0 early return

Reply:
also added a test case to confirm pop(0) is a no-op.